### PR TITLE
Document membership criteria and expectations

### PR DESCRIPTION
**Pull request by nikomatsakis**
This PR documents membership criteria and expectations, and discusses a simple pattern for selecting new leads.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
@rfcbot fcp merge

I think this is largely documenting existing practice, but I've also made a few changes (e.g., suggesting a brief write-up for a proposed new member). I'd like to get people to review and agree to the plans formally.

**Comment by nikomatsakis** [comment]
Also, I'd like to discuss leadership succession. @joshtriplett and I both agree that there should be some kind of regular check-in to decide how to proceed. I'm still happy leading the team, but I'm also interested in discussing with folks what it would look like for someone else to takeover and when that might make sense!

**Comment by nikomatsakis** [comment]
@rfcbot fcp merge

I think this is largely documenting existing practice, but I've also made a few changes (e.g., suggesting a brief write-up for a proposed new member). I'd like to get people to review and agree to the plans formally.

File: src/welcome.md [hunk 1]
```diff
@@ -6,7 +6,17 @@ as some amount of design constraints. It's always a
...
-Key links:
+## What is the lang team
+
+The lang team generally governs the "surface area" of the language, meaning both what code compiles and what happens when it executes. For new language features, we also assume a general "project management" role, in that we track the feature as it progresses from an idea, to implementation, and finally to stabilization.
+
+Note that implementation itself is governed by the [compiler team](https://github.com/rust-lang/compiler-team). To be very concrete, the lang team controls the spec, and the compiler team approves the implemenation and ensures that it meets the spec. Naturally, though, development is a collaborative process, and it often happens that the spec is altered in response to concerns that arise during implementation.
+
+We also work closely with the [types team](https://github.com/rust-lang/types-team), which owns the details of the type system design in much the same way as the compiler team owns the details of the implementation.
+
+Finally, there are often "grey areas" between the language and library teams, such as the addition of a new trait to the standard library that
```
Comment (scottmcm) [inline src/welcome.md:17]: Looks like this sentence trailed off at the

Comment (nikomatsakis) [reply src/welcome.md:17]: Actually, I think those commits are from #173 and I fixed it there

File: src/membership.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+* Promptly respond to RFC decisions by reviewing the question and either checking box or raising concerns
+
+## Process to add a new member
+
+Lang team members can propose new additions to the team as follows:
+
+* Lang team member prepares a short write-up to propose candidate to the rest of the team
+    * The write-up should draw on the qualifications below, giving examples where the candidate demonstrated the various criteria
+* Leads check with moderation team for known flags, surface to team if any.
+* The decision to add must have unanimous consent, similar to any other decision.
```
Comment (joshtriplett) [inline src/membership.md:30]:
```suggestion
* The decision to add must have unanimous consent, similar to any other decision. Objections may be raised in a private team discussion, or by contacting a team lead.
```

File: src/leads.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Lang team leadership
+
+Lang team leads drive the overall team agenda and tend to its well-being. They generally chair the meetings, keep an eye out for conflicts, and are to some extent the decision makers of last resort. They drive the roadmap process. They also post blog posts updating the public (possibly through delegation).
```
Comment (joshtriplett) [inline src/leads.md:3]:
```suggestion
Lang team leads drive the overall team agenda and tend to its well-being. They generally chair the meetings, keep an eye out for conflicts, ensure that tasks "someone" needs to do get a concrete owner, and help drive towards consensus. They coordinate the roadmap process regularly. They also take responsibility for posting blog posts updating the public (possibly through delegation).
```

File: src/membership.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Becoming and being a lang-team member
+
+Lang team members are the ones who ultimately decide the syntax/semantics of the Rust language. They judge whether a given piece of Rust syntax ought to compile and, if it does, what should happen when it runs.
+
+## Relations to other teams
+
+Lang team members should be familiar with how the compiler works, but they don't need to drive its implementation or even work on it. Compiler implementation details are the job of the [compiler team]. Note that compiler team members are encouraged to raise concerns if they feel that the desired design cannot be implemented in a reasoable and maintainable way.
```
Comment (joshtriplett) [inline src/membership.md:7]: Typo fix, and slight rephrasing.
```suggestion
Lang team members should be familiar with how the compiler works, but they don't need to drive its implementation or even work on it. Compiler implementation details are the job of the [compiler team]. Note that compiler team members can and should raise concerns if they feel that the desired design cannot be implemented in a reasonable and maintainable way.
```

File: src/membership.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+[compiler team]: https://rust-lang.github.io/compiler-team/
+
+Lang team members have to know the language inside and out, but they don't necessarily need deep type theory background. Working out the formal semantics of the language as well as diving into very detailed questions is covered by the [types team]. Types team members are encouraged to raise concerns if they feel that the desired design cannot be made sound or is internally inconsistent.
```
Comment (joshtriplett) [inline src/membership.md:11]:
```suggestion
Lang team members have to know the language inside and out, but they don't necessarily need deep type theory background. Working out the formal semantics of the language as well as diving into very detailed questions is covered by the [types team]. Types team members are encouraged to raise concerns if they feel that the desired design cannot be made sound, is internally inconsistent, or misses a potential point of greater orthogonality in design space.
```

File: src/membership.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+[types team]: https://rust-lang.github.io/types-team/
+
+## Expectations
+
+Lang team members are expected to
```
Comment (joshtriplett) [inline src/membership.md:17]:
```suggestion
Lang team members are expected to:
```

File: src/membership.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    * Have they led an impactful initiative to completion?
+    * Have they identified a criticial flaw in someone else's design that led to the design being altered?
+* Is this person **responsible**?
+    * When they agree to take on a task, do they either get it done or identify that they are not able to follow through and ask for help?
+* Is this person able to **lead others to a productive conversation**?
+    * Are there times when a conversation was stalled out and this person was able to step in and get the design discussion back on track?
+        * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
+* Is this person able to **disagree productively**?
+    * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
+    * Do they "steelman", looking for ways to restate others' points in the most convincing way?
```
Comment (joshtriplett) [inline src/membership.md:48]:
```suggestion
    * Do they "steelman", looking for ways to restate others' points in the most convincing way?
    * Do they demonstrate the ability to carefully introspect their own position and values, to seek common ground with others, and collaboratively work towards consensus?
```

File: src/membership.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    * When they agree to take on a task, do they either get it done or identify that they are not able to follow through and ask for help?
+* Is this person able to **lead others to a productive conversation**?
+    * Are there times when a conversation was stalled out and this person was able to step in and get the design discussion back on track?
+        * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
+* Is this person able to **disagree productively**?
+    * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
+    * Do they "steelman", looking for ways to restate others' points in the most convincing way? 
+* Is this person **active**?
+    * Are they attending the triage/design meetings regularly?
+    * Either in meeting or elsewhere, do they comment on disussions and otherwise?
```
Comment (joshtriplett) [inline src/membership.md:51]:
```suggestion
    * Either in meetings or elsewhere, do they participate in discussions?
```

File: src/membership.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
+* Is this person able to **disagree productively**?
+    * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
+    * Do they "steelman", looking for ways to restate others' points in the most convincing way?
+    * Do they demonstrate the ability to carefully introspect their own position and values, to seek common ground with others, and collaboratively work towards consensus?
+* Is this person **active**?
+    * Are they attending the triage/design meetings regularly?
+    * Either in meetings or elsewhere, do they participate in discussions?
+* Does this person have an **overall desire to improve the language**, rather than a strong interest in some particular domain?
+    * Everyone have preferences, but members are responsible for balancing a wide array of interests. Someone with very specialized interest may be a better choice for a lang team advisor.
```
Comment (nikomatsakis) [inline src/membership.md:54]:
```suggestion
    * Everyone has preferences, but members are responsible for balancing a wide array of interests. Someone with very specialized interest may be a better choice for a lang team advisor.
```

**Comment by joshtriplett** [comment]
LGTM, thanks for incorporating all the feedback.

**Pull request merged**